### PR TITLE
Add PinMAME mech component.

### PR DESCRIPTION
**Pull request by freezy**
This PR adds a Unity component that allows easily configuring a PinMAME custom mech handler.

It works by implementing a `IMechHandler` which contains a `OnMechUpdate` event that is triggered when new mech updates are received by PinMAME.

So we have a component that produces speed and positions. In order for the cannon to rotate, we need to convert this into an angle, and then rotate the geometry. I will document this more in detail in the main repo's PR, but in short there is a new `CannonComponent` that takes in the mech data and outputs the angle to the already existing `RotatorComponent`.

## TODO
- [x] Figure out how to handle `AddSw` methods
- [x] Error when trying to add more than n mechs
- [x] Create inspectors
- [x] Create icons
- [x] Finish [documentation](https://github.com/freezy/VisualPinball.Engine/blob/feature/mech/VisualPinball.Unity/Documentation~/plugins/pinmame/mechs.md) (shots with UI and some adaptions)

### CODE REVIEW

**Comment by freezy** [comment]
Sorry, there were many whitespace changes due to my IDE trimming lines on save. Check [this diff](https://github.com/VisualPinball/VisualPinball.Engine.PinMAME/pull/15/files?diff=unified&w=1) to ignore them.

**Comment by xantari** [comment]
@freezy Just watching the dev of this and am amazed. Love it. Curious, how are you finding out about these pinmame specifications?

File: VisualPinball.Engine.PinMAME.Unity/Runtime/PinMameEnums.cs [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+		/// <example>Sopranos pole dancers.</example>
+		OneSolenoid,
+
+		/// <summary>
+		/// Two solenoids, one for enabling the motor, one for setting the
+		/// direction.
+		/// </summary>
+		///
+		/// <example>The cannon in Terminator 2.</example>
+		OneDirectionalSolenoid,
```
Comment (jsm174) [inline VisualPinball.Engine.PinMAME.Unity/Runtime/PinMameEnums.cs:35]: T2 does not have two solenoids. 

4887a278e3da785a3dc60a08f9b58a564ce2ebb1

It works similar to this: https://www.youtube.com/embed/K7Pnxc1AncE

**Review by jsm174 — COMMENTED** [review]
Awesome stuff!

**Comment by jsm174** [comment]
Also, the most mechs we can support is 5, so thats why I added `_pinMame.GetMaxMechs()`. 

Not sure if you can implement this some how. `_pinMame.SetMech` will throw an exception if trying to set mech no >= 5.

https://github.com/vpinball/pinmame-dotnet/blob/master/src/PinMame/PinMame.cs#L722
https://github.com/vpinball/pinmame/blob/master/src/libpinmame/libpinmame.cpp#L809

File: VisualPinball.Engine.PinMAME.Unity/Runtime/PinMameMechComponent.cs [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+					switch (mark.Type) {
+						case PinMameMechSwitchMarkType.Switch:
+							mechConfig.AddSwitch(new PinMameMechSwitchConfig(mark.SwitchId, mark.StepBeginning, mark.StepEnd));
+							break;
+						case PinMameMechSwitchMarkType.PulseSwitch:
+							mechConfig.AddSwitch(new PinMameMechSwitchConfig(mark.SwitchId, mark.StepBeginning, mark.StepPulseDuration, 1));
+							break;
+						case PinMameMechSwitchMarkType.PulseSwitchNew:
+							mechConfig.AddSwitch(new PinMameMechSwitchConfig(mark.SwitchId, mark.StepBeginning, mark.StepEnd, mark.StepPulseDuration));
+							break;
```
[the remark this answers is not in the document]
Comment (freezy) [reply VisualPinball.Engine.PinMAME.Unity/Runtime/PinMameMechComponent.cs:124]: Yeah I don't like all those different variants, they are confusing to me. Will figure out something better.

File: VisualPinball.Engine.PinMAME.Unity/Runtime/PinMameEnums.cs (hunk 1 above)
Comment (freezy) [reply VisualPinball.Engine.PinMAME.Unity/Runtime/PinMameEnums.cs:35]: Okay so it runs only into one direction, but there is a physical mechanism that makes the cannon change direction after half a spin.

**Comment by freezy** [comment]
@jsm174 Yes, that's what I wanted to do actually but finally forgot.

@xantari I spent a couple of evenings looking at table scripts, PinMAME code and forum posts. Also @jsm174 took the time to explain a few things :)

**Pull request merged**